### PR TITLE
Feature hw 5

### PR DESCRIPTION
**Pull request by kristina-salnyk**

### CODE REVIEW

File: src/common/Loader/Loader.jsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { RotatingLines } from 'react-loader-spinner';
 
 import { LoaderWrap } from './Loader.styled';
 
-const Loader = ({ className }) => {
+const Loader = ({ width = '100', className }) => {
```
Comment (obriukhanov) [inline src/common/Loader/Loader.jsx]: Best practice:
Variables with default assignment should be settled after usual variables.

File: src/components/CourseForm/tests/courseForm.test.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+const mockedAuthorName = 'Test Author';
+
+jest.mock('react-router-dom', () => ({
+	...jest.requireActual('react-router-dom'),
+	useNavigate: () => jest.fn(),
+	useLocation: () => jest.fn(),
+}));
+
+test('should show all authors list', () => {
```
Comment (obriukhanov) [inline src/components/CourseForm/tests/courseForm.test.js]: Here you can wrap test cases into 'describe' block.
And in it's scope you can place some reusable logic.
Here in each test case you render CourseFrom.
You can create necessary variables in describe block and restore/clear mocks in before/afterEach if needed.
It'll decrease code duplications.

File: src/components/CourseInfo/CourseInfo.jsx [hunk 3]
```diff
@@ -29,18 +30,41 @@ import {
...
+		dispatch(fetchAuthors(setIsLoading));
+	}, [dispatch]);
+
+	useEffect(() => {
+		if (courseDataFetched.current) return;
+		courseDataFetched.current = true;
+
+		(async () => {
+			const result = await dispatch(fetchCourse(courseId, setIsLoading));
+			if (result.successful) setCourse({ ...result.data });
```
Comment (obriukhanov) [inline src/components/CourseInfo/CourseInfo.jsx]: Why spread?

File: src/components/Courses/Courses.jsx [hunk 4]
```diff
@@ -20,45 +24,73 @@ import {
...
+								navigate(ROUTES.CREATE_COURSE);
+							}}
 						/>
-					</CoursesMessage>
-				)}
+					)}
+				</CoursesHeader>
+				{isLoading && <LoaderStyled />}
+				{!isLoading &&
+					(courses.length > 0 ? (
```
Comment (obriukhanov) [inline src/components/Courses/Courses.jsx]: For better readability create a new constant.
E.g canRenderCourseList.
I don't recommend use in JSX 2-leveled conditions.

File: src/store/courses/thunk.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+	try {
+		changeIsLoading(true);
+
+		const response = await getCourse(courseId);
+		const { data } = response;
+
+		if (response.status === 200 && data.successful) {
+			const courseObj = data.result;
+			result.successful = true;
+			result.data = courseObj;
```
Comment (obriukhanov) [inline src/store/courses/thunk.js:159]: Just FYI
Here is a useful trick if you want conditionally create object with come new field and not mutate it.
Can be useful in future.
const courseObj = {
...data.result,
...(response.status === 200 && data.successful) && {successful: true, ... etc})
}

**Review by obriukhanov — APPROVED** [review]
LGTM

File: src/common/Loader/Loader.jsx (hunk 1 above)
Comment (kristina-salnyk) [reply src/common/Loader/Loader.jsx]: The order of the variables has been changed

File: src/components/CourseForm/tests/courseForm.test.js (hunk 2 above)
Comment (kristina-salnyk) [reply src/components/CourseForm/tests/courseForm.test.js]: The repeatable logic has been moved to the description block

File: src/components/CourseInfo/CourseInfo.jsx (hunk 3 above)
Comment (kristina-salnyk) [reply src/components/CourseInfo/CourseInfo.jsx]: The Spread operator has been removed

File: src/components/Courses/Courses.jsx (hunk 4 above)
Comment (kristina-salnyk) [reply src/components/Courses/Courses.jsx]: Edits to rendering conditions have been added

File: src/store/courses/thunk.js (hunk 5 above)
Comment (kristina-salnyk) [reply src/store/courses/thunk.js:159]: Done

**Pull request merged**